src/corelibs/U2Algorithm/src/phyltree/PhyTreeGeneratorTask.cpp

```diff
@@ -29,91 +29,70 @@
     tpm = Task::Progress_Manual;
 }
 
-void PhyTreeGeneratorTask::run() {
+const PhyTree& PhyTreeGeneratorTask::getResult() const {
+    return result;
 }
 
-Task::ReportResult PhyTreeGeneratorTask::report() {
-    return ReportResult_Finished;
+const CreatePhyTreeSettings& PhyTreeGeneratorTask::getSettings() const {
+    return settings;
 }
 
 PhyTreeGeneratorLauncherTask::PhyTreeGeneratorLauncherTask(const MultipleSequenceAlignment& ma, const CreatePhyTreeSettings& _settings)
-    : Task(PhyTreeGeneratorLauncherTask::tr("Calculating Phylogenetic Tree"), TaskFlag_FailOnSubtaskError), inputMA(ma->getCopy()), settings(_settings), task(nullptr) {
+    : Task(PhyTreeGeneratorLauncherTask::tr("Calculating Phylogenetic Tree"), TaskFlag_NoRun | TaskFlag_FailOnSubtaskError),
+      inputMA(ma->getCopy()), settings(_settings), task(nullptr) {
     tpm = Task::Progress_SubTasksBased;
 }
-void PhyTreeGeneratorLauncherTask::prepare() {
-    QString algId = settings.algorithm;
-    PhyTreeGeneratorRegistry* registry = AppContext::getPhyTreeGeneratorRegistry();
-    PhyTreeGenerator* generator = registry->getGenerator(algId);
-    if (generator == nullptr) {
-        stateInfo.setError(PhyTreeGeneratorLauncherTask::tr("Tree construction algorithm %1 not found").arg(algId));
-    } else {
-        const QStringList& rowsOrder = settings.rowsOrder;
-        if (rowsOrder.size() >= inputMA->getRowNames().size()) {
-            inputMA->sortRowsByList(rowsOrder);
-        }
 
-        namesConvertor.replaceNamesWithAlphabeticIds(inputMA);
+static const QString RENAMED_ROW_PREFIX = "r";
 
-        task = dynamic_cast<PhyTreeGeneratorTask*>(generator->createCalculatePhyTreeTask(inputMA, settings));
-        addSubTask(task);
+void PhyTreeGeneratorLauncherTask::prepare() {
+    PhyTreeGeneratorRegistry* registry = AppContext::getPhyTreeGeneratorRegistry();
+    PhyTreeGenerator* generator = registry->getGenerator(settings.algorithm);
+    CHECK_EXT(generator != nullptr, stateInfo.setError(PhyTreeGeneratorLauncherTask::tr("Tree algorithm %1 is not found").arg(settings.algorithm)), );
+
+    // Assign unique names to rows. The row name is a string representation of the index + 'row_' prefix.
+    originalRowNameByIndex = inputMA->getRowNames();
+    int rowsCount = inputMA->getRowCount();
+    for (int i = 0; i < rowsCount; i++) {
+        QString uniqueRowName = RENAMED_ROW_PREFIX + QString::number(i);
+        inputMA->renameRow(i, uniqueRowName);
     }
+    task = qobject_cast<PhyTreeGeneratorTask*>(generator->createCalculatePhyTreeTask(inputMA, settings));
+    SAFE_POINT(task != nullptr, "Not a PhyTreeGeneratorTask!", );
+    addSubTask(task);
 }
+
 Task::ReportResult PhyTreeGeneratorLauncherTask::report() {
-    if (task) {
-        result = task->getResult();
-        namesConvertor.restoreNames(result);
+    CHECK(!stateInfo.isCoR() && task != nullptr && !task->getStateInfo().isCoR(), ReportResult_Finished);
```

Comment:
`SAFE_POINT_EXT(task != nullptr, setError("PhyTreeGeneratorTask got lost"), ReportResult_Finished);`

This would have much sence

Reply:
Nope, canceled task is not a SAFE_POINT (failure)